impl.go

```diff
@@ -447,7 +459,21 @@ to prevent shell globbing.
 		}
 	}
 
-	fns, err := funcs(iface, *flagSrcDir, EmitComments(*flagComments))
+	var recvPkg = *flagRecvPkg
+	if recvPkg == "" {
+		//  "   s *Struct   " , receiver: Struct
+		recvs := strings.Fields(recv)
+		receiver := recvs[len(recvs)-1] // note that this correctly handles "s *Struct" and "*Struct"
+		receiver = strings.TrimPrefix(recv, "*")
+		pkg, _, err := typeSpec("", receiver, *flagSrcDir)
+		if err == nil {
+			recvPkg = pkg.Package.Name
+		} else {
+			*flagRecvPkg = "" // Cannot be matched with any package
```

Comment:
On a second read, I believe that this entire else clause can be deleted. The only way to get here is if `recvPkg` is already the empty string. So we can simply do nothing here.